pandora-common/src/networking/directory_client.ts

```diff
@@ -101,6 +101,40 @@ export const ACCOUNT_SETTINGS_DEFAULT = Object.freeze<IDirectoryAccountSettings>
 	interfaceChatroomOfflineCharacterFilter: 'ghost',
 });
 
+export const DirectoryAccountSettingsLimitedSchema = z.object({
+	displayName: DisplayNameSchema,
+});
+export type IDirectoryAccountSettingsLimited = z.infer<typeof DirectoryAccountSettingsLimitedSchema>;
+
+type DirectoryAccountSettingsLimitedStoredSchemaAcc = {
```

Comment:
Optional suggestion:
This complexity could be avoided altogether, if the "next allowed change" data was next to the values instead of interleaved with them, i.e.:
```ts
type IDirectoryAccountInfo = {
    // ...
    settingsLimited: IDirectoryAccountSettingsLimited;
    settingsLimitedCooldowns: Partial<Record<keyof IDirectoryAccountSettingsLimited, number>>;
    // ...
}
```

Reply:
Actually thinking about it more...
Why is this a completely separate settings framework in the first place when the following would have been enough to add limited settings _if we actually want them in the first place_?
```ts
// Add the `displayName` to the existing settings

const ACCOUNT_SETTINGS_LIMITS: Partial<Record<keyof IDirectoryAccountSettings, number>> = {
    displayName: TimeSpanMs(1, 'months'),
};

type IDirectoryAccountInfo = {
    // ...
    settingsCooldowns: Partial<Record<keyof IDirectoryAccountSettings, number>>;
    // ...
}
```